app/questionnaire/placeholder_parser.py

```diff
@@ -97,8 +82,11 @@ def _parse_transforms(self, transform_list: Sequence[Mapping]):
 
         for transform in transform_list:
             transform_args: Dict[str, Union[None, str, List[str]]] = {}
+
             for arg_key, arg_value in transform["arguments"].items():
-                if not isinstance(arg_value, dict):
+                if isinstance(arg_value, list):
+                    transform_args[arg_key] = self._resolve_value_source_list(arg_value)
+                elif not isinstance(arg_value, dict):
```

Comment:
I think it would be clearer if we made this more explicit by removing the `not`, something like:

```python
if isinstance(arg_value, list):
    transform_args[arg_key] = self._resolve_value_source_list(arg_value)
elif isinstance(arg_value, dict):
    if "value" in arg_value:
        transform_args[arg_key] = arg_value["value"]
    elif arg_value["source"] == "previous_transform":
        transform_args[arg_key] = transformed_value
    else:
        transform_args[arg_key] = self._resolve_value_source(arg_value)
else:
    transform_args[arg_key] = arg_value

```